SYCL/ESIMD/api/functional/mutator.hpp

```diff
@@ -0,0 +1,115 @@
+//===-- mutator.hpp - This file provides common function and classes to mutate
+//      reference data. ---------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides common function and classes to mutate reference data.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "type_traits.hpp"
+#include "value.hpp"
+#include <sycl/sycl.hpp>
+
+// for std::for_each
+#include <algorithm>
+#include <vector>
+
+namespace esimd_test::api::functional {
+
+// In some test cases it’s possible to pass a reference data as the input data
+// directly without any modification. For example, if we check some copy
+// constructor or memory move operation it’s OK to have any data values as the
+// input ones. But in most cases we need to consider the possibility of UB for
+// C++ operations due to modification of input values.
+//
+// Mutation mechanism covers such requirement.
+// The mutator namespace is intended to store such mutators alongside with
+// the generic ones provided below.
+namespace mutator {
+
+// Replace specific reference values to the bigger ones, so we can safely
+// substract `val` later.
```

Comment:
```suggestion
// substract `m_value` later.
```
We have multiple `val` here, but single `m_value`

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/789/commits/41df5e531f17caed6253334af281f97f021058d3#diff-fa18495d278b576095ad26f3fbce0799c62e2302a8837d4e0a289c922d0dce74R39) and in other cases